999-aws-ebs-nvme.rules

```diff
@@ -1 +1 @@
-ACTION=="add", SUBSYSTEM=="block", KERNEL=="nvme[1-26]n1", ATTRS{model}=="Amazon Elastic Block Store              ", RUN+="/usr/local/bin/ebs-nvme-mapping"
+SUBSYSTEM=="block", KERNEL=="nvme[0-9]*n[0-9]*", ATTRS{model}=="Amazon Elastic Block Store", PROGRAM+="/usr/local/sbin/ebs-nvme-mapping.sh /dev/%k" SYMLINK+="%c"
```

Comment:
Why did we drop `ACTION` from the rule?

The valid block device names that are matched should not be unnecessarily widened. The official “spec” is stated at https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html

But I am open to convincing as to why it’s a good idea to run wider than the spec and include the root NVME device in our machinations.

I very much like the use of the kernel name and program result string substitutions.

Reply:
> Why did we drop `ACTION` from the rule?

This was a change originally made by @brahman81 and included in my repo. I suspect it is because we want the rule to be run on both add and remove, removing the symlink when the volume is removed. I'll do some more testing and verify this behavior. I believe we could also use `action=="add|remove"`.

> The valid block device names that are matched should not be unnecessarily widened. The official “spec” is stated at https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/device_naming.html

Noted and updated.

Reply:
I've verified on Ubuntu 18.04 that having no action set works for both add and remove. Using `action=="add|remove"` did _not_ work.